src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -145,9 +148,10 @@ public Command sysIdDynamic(SysIdRoutine.Direction direction) {
 
   @Override
   public void periodic() {
+    encoderDegrees = calculateDegrees();
```

Comment:
Instead of retrieving the latest encoder value every single time it's needed, we can access it once here in `periodic()` and reuse that stored value as many times as we need in the loop. Getting the encoder value doesn't take long, but when called multiple times in a single loop it can add up.